File: 2582-pass-the-pillow/2582-pass-the-pillow.php
class Solution {

    /**
     * @param Integer $n
     * @param Integer $time
     * @return Integer
     */
    function passThePillow($n, $time) {
        // remove full cycles
        $roundTrip = ($n-1) * 2;
        $time %= $roundTrip;

        // left to right
        if($time < $n) return $time + 1;

        // right to left
        $time -= $n;
        return $n - $time - 1;
    }
}

File: 2582-pass-the-pillow/2582-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
        // remove full cycles
        var roundTrip = (n-1)*2;
        time %= roundTrip;

        // left to right
        if(time < n) return time+1;
        
        // right to left
        time -= n;
        return n - time - 1;
    }
}